src/lib/utils/fixFingerprintDataTypes.ts

```diff
@@ -0,0 +1,24 @@
+import { IFingerprint } from "../../types";
+import { fingerprintNoQuotes } from "../../types/Fingerprint";
+
+export function fixFingerprintDataTypes(fingerprint: IFingerprint): IFingerprint {
+    const newFp = (fingerprint as unknown) as Record<string, unknown>;
+
+    // set correct data types (int are intigers, bool are booleans)
+    Object.entries(fingerprint).forEach(([key, value]) => {
+        if (value == null || value == "null") {
+            newFp[key] = null;
+        } else if (fingerprintNoQuotes.includes(key) && value.toLowerCase !== undefined) {
+            if (value.toLowerCase() == "false") {
+                newFp[key] = false;
+            } else if (value.toLowerCase() == "true") {
+                newFp[key] = true;
```

Comment:
We can convert this into a single `if statement`
```suggestion
            if (["true", "false"].includes(value.toLowerCase())) {
                newFp[key] = value.toLowerCase() === "true";
```